### PR TITLE
Expose all references when not restoring

### PR DESCRIPTION
**Pull request by dougbu**
- use empty `$(MSBuildRestoreSessionId)` to determine when contributing to dependency graph

- remove extra direct references
  - should now be part of the dependency graph automatically

### CODE REVIEW

File: src/SignalR/common/SignalR.Common/src/Microsoft.AspNetCore.SignalR.Common.csproj [hunk 1]
```diff
@@ -25,11 +25,12 @@
     <Reference Include="Microsoft.Extensions.Options" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == '$(DefaultNetFxTargetFramework)'">
+  <ItemGrou
```
Comment (BrennanConroy) [inline src/SignalR/common/SignalR.Common/src/Microsoft.AspNetCore.SignalR.Common.csproj:28]:
```suggestion
  <ItemGroup
```

File: src/SignalR/common/SignalR.Common/src/Microsoft.AspNetCore.SignalR.Common.csproj [hunk 2]
```diff
@@ -25,11 +25,12 @@
     <Reference Include="Microsoft.Extensions.Options" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == '$(DefaultNetFxTargetFramework)'">
+  <ItemGroup
+       Condition="'$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == '$(DefaultNetFxTargetFramework) OR '$(MSBuildRestoreSessionId)' == ''">
```
Comment (dougbu) [inline src/SignalR/common/SignalR.Common/src/Microsoft.AspNetCore.SignalR.Common.csproj:29]:
```suggestion
       Condition="'$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == '$(DefaultNetFxTargetFramework)' OR '$(MSBuildRestoreSessionId)' == ''">
```

**Comment by dougbu** [comment]
- #31075

File: src/SignalR/common/SignalR.Common/src/Microsoft.AspNetCore.SignalR.Common.csproj [hunk 3]
```diff
@@ -25,11 +25,12 @@
     <Reference Include="Microsoft.Extensions.Options" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == '$(DefaultNetFxTargetFramework)'">
+  <ItemGrou
+       Condition="'$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == '$(DefaultNetFxTargetFramework) OR '$(MSBuildRestoreSessionId)' == ''">
```
Comment (dougbu) [inline src/SignalR/common/SignalR.Common/src/Microsoft.AspNetCore.SignalR.Common.csproj:29]:
```suggestion
  <ItemGroup
       Condition="'$(TargetFramework)' == 'netstandard2.0' OR '$(TargetFramework)' == '$(DefaultNetFxTargetFramework)' OR '$(MSBuildRestoreSessionId)' == ''">
```

**Review by wtgodbe — APPROVED** [review]

**Comment by dougbu** [comment]
@akoeplinger something strange is going on. The action is failing even after removing differences between this repo and dotnet/runtime in .github/workflows/backport.yml and eng/actions/backport/*, See #32777 then #32927.

I also verified https://octokit.github.io/rest.js/v18#issues-create-comment still matches what we're doing in this action.

Any ideas why either the `issues` property is `undefined` or GitHub's JS parser is interpreting `createComment(...)` as a property reference❔

**Comment by dougbu** [comment]
@akoeplinger I figured it out. Need to use the `rest` object in one more place. See #32930. I'll put up a similar dotnet/runtime PR on Monday.

The other part of the issue here was conflicts when the action attempted to apply the patch. Anyone know if `git am --3way` would help create a PR containing the conflicts❔

**Comment by dougbu** [comment]
> Anyone know if `git am --3way` would help create a PR containing the conflicts❔

Oops, index.js already uses `--3way`. Probably need more complicated `git` hacks to create a PR despite the conflicts.

**Comment by dougbu** [comment]
Let's try this again&hellip;

**Comment by Pilchie** [comment]
FYI - I *think* I just changed @msftbot to not comment when @github-actions[bot] comments on a closed PR.

**Pull request merged**